--- a/command.c
+++ b/command.c
@@ -473,12 +473,31 @@
                   (long long)(movie->frame_counter));
       }
       else
          strlcpy(value_dynamic, "0 0 0", sizeof(value_dynamic));
    }
    #endif
+#ifdef HAVE_MENU
+   else if (string_is_equal(arg, "menu_active"))
+   {
+      struct menu_state* menu_st = menu_state_get_ptr();
+      if (menu_st && (menu_st->flags & MENU_ST_FLAG_ALIVE))
+         value = "true";
+      else
+         value = "false";
+   }
+#endif
+#ifdef HAVE_CHEEVOS
+   else if (string_is_equal(arg, "cheevos_enable"))
+   {
+      if (settings->bools.cheevos_enable)
+         value = "true";
+      else
+         value = "false";
+   }
+#endif
    /* TODO: query any string */
 
    _len  = strlcpy(reply, "GET_CONFIG_PARAM ", sizeof(reply));
    _len += strlcpy(reply + _len, arg, sizeof(reply)  - _len);
    reply[  _len] = ' ';
    reply[++_len] = '\0';
@@ -764,12 +783,39 @@
    else
       ret = false;
 
    cmd->replier(cmd, reply, _len);
    return ret;
 }
+
+bool command_save_state_slot(command_t* cmd, const char* arg)
+{
+   char state_path[PATH_MAX_LENGTH] = "";
+   size_t _len                  = 0;
+   char reply[128]              = "";
+   unsigned int slot            = (unsigned int)strtoul(arg, NULL, 10);
+   bool savestates_enabled      = core_info_current_supports_savestate();
+   bool ret                     = false;
+   _len  = strlcpy(reply, "SAVE_STATE_SLOT ", sizeof(reply));
+   _len += snprintf(reply + _len, sizeof(reply) - _len, "%d", slot);
+   if (savestates_enabled)
+   {
+      size_t info_size;
+      runloop_get_savestate_path(state_path, sizeof(state_path), slot);
+
+      info_size          = core_serialize_size();
+      savestates_enabled = (info_size > 0);
+   }
+   if (savestates_enabled)
+      ret = content_save_state(state_path, true);
+   else
+      ret = false;
+
+   cmd->replier(cmd, reply, _len);
+   return ret;
+}
 
 bool command_play_replay_slot(command_t *cmd, const char *arg)
 {
 #ifdef HAVE_BSV_MOVIE
    char replay_path[16384];
    char reply[128]              = "";
--- a/command.h
+++ b/command.h
@@ -424,12 +424,13 @@
 
 bool command_version(command_t *cmd, const char* arg);
 bool command_get_status(command_t *cmd, const char* arg);
 bool command_get_config_param(command_t *cmd, const char* arg);
 bool command_show_osd_msg(command_t *cmd, const char* arg);
 bool command_load_state_slot(command_t *cmd, const char* arg);
+bool command_save_state_slot(command_t* cmd, const char* arg);
 bool command_play_replay_slot(command_t *cmd, const char* arg);
 bool command_seek_replay(command_t *cmd, const char *arg);
 bool command_save_savefiles(command_t *cmd, const char* arg);
 bool command_load_savefiles(command_t *cmd, const char* arg);
 #ifdef HAVE_CHEEVOS
 bool command_read_ram(command_t *cmd, const char *arg);
@@ -454,12 +455,13 @@
    { "WRITE_CORE_RAM",   command_write_ram,        "<address> <byte1> <byte2> ..." },
 #endif
    { "READ_CORE_MEMORY", command_read_memory,      "<address> <number of bytes>" },
    { "WRITE_CORE_MEMORY",command_write_memory,     "<address> <byte1> <byte2> ..." },
 
    { "LOAD_STATE_SLOT",command_load_state_slot, "<slot number>"},
+   { "SAVE_STATE_SLOT",command_save_state_slot, "<slot number>"},
    { "PLAY_REPLAY_SLOT",command_play_replay_slot, "<slot number>"},
    { "SEEK_REPLAY",command_seek_replay, "<frame number>"},
 
    { "SAVE_FILES", command_save_savefiles, "No argument"},
    { "LOAD_FILES", command_load_savefiles, "No argument"},
 
